core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/udf/UserDFOperator.java

```diff
@@ -0,0 +1,322 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.uci.ics.texera.api.constants.ErrorMessages;
+import edu.uci.ics.texera.api.constants.DataConstants.TexeraProject;
+import edu.uci.ics.texera.api.dataflow.IOperator;
+import edu.uci.ics.texera.api.dataflow.ISourceOperator;
+import edu.uci.ics.texera.api.exception.DataflowException;
+import edu.uci.ics.texera.api.exception.TexeraException;
+import edu.uci.ics.texera.api.schema.AttributeType;
+import edu.uci.ics.texera.api.schema.Schema;
+import edu.uci.ics.texera.api.tuple.Tuple;
+import edu.uci.ics.texera.api.utils.Utils;
+import edu.uci.ics.texera.dataflow.common.AbstractSingleInputOperator;
+import sun.misc.Signal;
+import sun.misc.SignalHandler;
+
+/**
+ * @author Qinhua Huang
+ */
+
+public class UserDFOperator extends AbstractSingleInputOperator implements SignalHandler, ISourceOperator{
+    private UserDFOperatorPredicate predicate;
+    private Schema outputSchema;
+    
+    /* 2 input signal:
+     *      TAG_NULL= "": result NULL
+     *      TAG_LEN = length of text
+     * 3 ouput signal:
+     *      TAG_NULL= "": result NULL
+     *      TAG_WAIT= Character.unsigned:   need to wait for next
+     *      TAG_LEN = length of text
+     */
+    // Signal write to buffer
+    int POSITION_PID = 0;
+    int POSITION_TAG = 10;
+    int POSITION_JSON = 20;
+    
+    String TAG_NULL = "0\n";
+    String TAG_WAIT = "w";
+    String TAG_LEN;
+    
+    //Signal used between processes
+    int IPC_SIG = 12;
+    String IPC_SIG_STRING = "USR2";
+    
+    // Named buffer in file system
+    
+    private String inputFilePath = Utils.getResourcePath("input_java_python.txt", TexeraProject.TEXERA_DATAFLOW).toString();
+    private String outputFilePath = Utils.getResourcePath("output_java_python.txt", TexeraProject.TEXERA_DATAFLOW).toString();
+    private String pythonDebugOutputFilePath = Utils.getResourcePath("debug_output_python.txt", TexeraProject.TEXERA_DATAFLOW).toString();
+    final int mmapBufferSize = 1024* 8;
+    
+    private String PYTHON = "python3";
+    private String PYTHONSCRIPT = Utils.getResourcePath("udf_operator.py", TexeraProject.TEXERA_DATAFLOW).toString();
+    private boolean getPythonResult = false;
+    public Process processPython;
+    private String pythonPID;
+    
+    public Tuple outputTuple;
+    
+    private MappedByteBuffer inputBuffer;
+    private MappedByteBuffer outputBuffer;
+    
+    private FileChannel inputFileChannel;
+    private FileChannel outputFileChannel;
+    
+    public UserDFOperator(UserDFOperatorPredicate predicate) {
+        this.predicate = predicate;
+        this.pythonPID = null;
+        this.inputBuffer = null;
+        this.outputBuffer = null;
+    }
+    
+    private String getJavaPID() {
```

Comment:
For functions like these where we get an output from the system/sore other function and do some formatting, we should write a sample input in the comments.